src/@utils/order.ts

```diff
@@ -81,33 +86,33 @@ export async function order(
       consumeMarketFeeAddress: marketFeeAddress,
       consumeMarketFeeAmount: consumeMarketOrderFee,
       consumeMarketFeeToken:
-        asset?.accessDetails?.baseToken?.address ||
+        accessDetails.baseToken?.address ||
         '0x0000000000000000000000000000000000000000'
     }
   } as OrderParams
 
-  switch (asset.accessDetails?.type) {
+  switch (accessDetails.type) {
     case 'fixed': {
       // this assumes all fees are in ocean
 
       const freParams = {
         exchangeContract: config.fixedRateExchangeAddress,
-        exchangeId: asset.accessDetails.addressOrId,
+        exchangeId: accessDetails.addressOrId,
         maxBaseTokenAmount: orderPriceAndFees.price,
-        baseTokenAddress: asset?.accessDetails?.baseToken?.address,
-        baseTokenDecimals: asset?.accessDetails?.baseToken?.decimals || 18,
+        baseTokenAddress: accessDetails.baseToken?.address,
+        baseTokenDecimals: accessDetails.baseToken?.decimals || 18,
         swapMarketFee: consumeMarketFixedSwapFee,
         marketFeeAddress
       } as FreOrderParams
 
-      if (asset.accessDetails.templateId === 1) {
+      if (accessDetails.templateId === 1) {
```

Comment:
also here enum?

Reply:
Honestly I don't know what are the options for templateId. I think it's all over the codebase and it's something that we would have to check later